core/app/models/spree/calculator/weighted_rate.rb

```diff
@@ -0,0 +1,19 @@
+require_dependency 'spree/calculator'
+
+module Spree
+  class Calculator::WeightedRate < Calculator
+    preference :amount, :decimal, default: 0
+    preference :currency, :string, default: -> { Spree::Config[:currency] }
+
+    def compute_line_item(line_item)
+      if line_item && preferred_currency.casecmp(line_item.currency).zero?
```

Comment:
Aahhh I gotcha. Right now `line_item.currency` actually delegates to the order already: https://github.com/solidusio/solidus/blob/master/core/app/models/spree/line_item.rb#L41